packages/platform-browser/src/dom/shared_styles_host.ts

```diff
@@ -72,6 +106,7 @@ export class DomSharedStylesHost extends SharedStylesHost implements OnDestroy {
 
   ngOnDestroy(): void {
     this._hostNodes.forEach(styleNodes => styleNodes.forEach(removeStyle));
+    this._styleNodesInDOM?.forEach(e => e.remove());
```

Comment:
I think we can also clear the map on destroy:
```suggestion
    this._styleNodesInDOM?.forEach(e => e.remove());
    this._styleNodesInDOM?.clear();
```